src/fs.ts

```diff
@@ -34,45 +34,54 @@ export class RemoteFileSystem implements FileSystem {
 	 * The files request is sent from the server to the client to request a list of all files in the workspace or inside the directory of the base parameter, if given.
 	 * A language server can use the result to index files by filtering and doing a content request for each text document of interest.
 	 */
-	async getWorkspaceFiles(base?: string, childOf = new Span()): Promise<Iterable<string>> {
-		return iterate(await this.client.workspaceXfiles({ base }, childOf))
-			.map(textDocument => textDocument.uri);
+	async getWorkspaceFiles(base?: URL, childOf = new Span()): Promise<Iterable<URL>> {
+		return iterate(await this.client.workspaceXfiles({ base: base && base.href }, childOf))
+			.map(textDocument => new URL(textDocument.uri));
 	}
 
 	/**
 	 * The content request is sent from the server to the client to request the current content of any text document. This allows language servers to operate without accessing the file system directly.
 	 */
-	async getTextDocumentContent(uri: string, childOf = new Span()): Promise<string> {
-		const textDocument = await this.client.textDocumentXcontent({ textDocument: { uri } }, childOf);
+	async getTextDocumentContent(uri: URL, childOf = new Span()): Promise<string> {
+		const textDocument = await this.client.textDocumentXcontent({ textDocument: { uri: uri.href } }, childOf);
 		return textDocument.text;
 	}
 }
 
+/**
+ * FileSystem implementation that reads from the local disk
+ */
 export class LocalFileSystem implements FileSystem {
 
 	/**
-	 * @param rootPath The root directory path that relative URIs should be resolved to
+	 * @param rootUri The workspace root URI that is used when no base is given
 	 */
-	constructor(private rootPath: string) {}
+	constructor(protected rootUri: URL) {}
 
 	/**
-	 * Converts the URI to an absolute path
+	 * Returns the file path where a given URI should be located on disk
 	 */
-	protected resolveUriToPath(uri: string): string {
-		return toUnixPath(path.resolve(this.rootPath, uri2path(uri)));
+	protected resolveUriToPath(uri: URL): string {
+		return uri2path(uri);
```

Comment:
Tried it (if you look at commit log) but it is useful for overriding it. Otherwise the globbing etc. needs to be completely reimplemented.